src/main/scala/com/databricks/spark/csv/CsvParser.scala

```diff
@@ -16,24 +16,24 @@
 package com.databricks.spark.csv
 
 
-import org.apache.spark.sql.{DataFrame, SQLContext}
 import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.{DataFrame, SQLContext}
+
 import com.databricks.spark.csv.util.{ParserLibs, ParseModes, TextFile}
 
 /**
  * A collection of static functions for working with CSV files in Spark SQL
  */
 class CsvParser {
-
   private var useHeader: Boolean = false
-  private var delimiter: Character = ','
-  private var quote: Character = '"'
-  private var escape: Character = null
+  private var csvParsingOpts: CSVParsingOpts = CSVParsingOpts()
```

Comment:
as the parser improves a ton of options are being added. it makes sense to organize them in containers for ease of use. imho, schema is a separate thing. i intended csvparsingopts to contain options that are pass to the csv parsing library. i can change the name if there is a better one you can think of